src/Components/ServiceScene/Breakdowns/PatternsViewTextSearch.tsx

```diff
@@ -0,0 +1,155 @@
+              patternsBreakdownScene.state.patternFilter,
+              this.onSearchResult
+            );
+          }
+        }
+      })
+    );
+
+    this._subs.add(
+      patternsBreakdownScene.subscribeToState((newState, prevState) => {
+        // If we have a search string, but no filtered patterns, run the search
+        if (
+          newState.patternFilter &&
+          !newState.filteredPatterns &&
+          newState.patternFrames &&
+          JSON.stringify(newState.filteredPatterns) !== JSON.stringify(prevState.filteredPatterns)
+        ) {
+          fuzzySearch(
+            newState.patternFrames.map((frame) => frame.pattern),
+            newState.patternFilter,
+            this.onSearchResult
+          );
+        }
+      })
+    );
+  }
+
+  /**
+   * Sets the patterns filtered by string match
+   * @param patterns
+   * @param patternFramesOverride
+   */
+  setFilteredPatterns(patterns: string[], patternFramesOverride?: PatternFrame[]) {
+    const patternsBreakdownScene = sceneGraph.getAncestor(this, PatternsBreakdownScene);
+    const patternFrames = patternFramesOverride ?? patternsBreakdownScene.state.patternFrames;
+
+    if (patternFrames) {
+      const filteredPatternFrames = patternFrames.filter((patternFrame) => {
+        if (patternsBreakdownScene.state.patternFilter && patternFrames?.length) {
+          return patterns.find((pattern) => pattern === patternFrame.pattern);
+        }
+        return false;
+      });
+
+      patternsBreakdownScene.setState({
+        filteredPatterns: filteredPatternFrames,
+      });
+    }
+  }
+
+  /**
+   * Fuzzy search callback
+   * @param data
+   */
+  onSearchResult = (data: string[][]) => {
+    const patternsBreakdownScene = sceneGraph.getAncestor(this, PatternsBreakdownScene);
+    // If we have a search string
+    if (patternsBreakdownScene.state.patternFilter) {
+      this.setFilteredPatterns(data[0]);
+    } else if (patternsBreakdownScene.state.filteredPatterns && !patternsBreakdownScene.state.patternFilter) {
+      // Wipe the parent filtered state
+      this.setEmptySearch();
+    }
+  };
+
+  /**
+   * Wipes filtered patterns when search string is empty
+   */
+  private setEmptySearch() {
+    const parent = sceneGraph.getAncestor(this, PatternsBreakdownScene);
+    parent.setState({
+      filteredPatterns: undefined,
+    });
+  }
+}
+
+const styles = {
+  field: css({
+    label: 'field',
+    marginBottom: 0,
+  }),
+  icon: css({
+    cursor: 'pointer',
+  }),
+};
+
+export function PatternTextSearchComponent({ model }: SceneComponentProps<PatternsViewTextSearch>) {
+  const patternsBreakdownScene = sceneGraph.getAncestor(model, PatternsBreakdownScene);
+  const { patternFilter } = patternsBreakdownScene.useState();
+  return (
+    <Field className={styles.field}>
+      <Input
+        suffix={<Icon onClick={model.clearSearch} className={styles.icon} name={'x'} />}
```

Comment:
```suggestion
        suffix={<Icon onClick={model.clearSearch} className={styles.icon} name={'times'} />}
```

times should be used to cancel.

Reply:
Actually, this can use the `SearchInput` component from main now.

Reply:
Yeah good catch, I swapped it out!